src/librustc_mir/transform/generator.rs

```diff
@@ -567,6 +556,34 @@ fn renumber_bitset(
     out
 }
 
+/// Record conflicts between locals at the current dataflow cursor positions.
+///
+/// You need to seek the cursors before calling this function.
+fn record_conflicts_at_curr_loc(
+    local_conflicts: &mut BitMatrix<Local, Local>,
+    init: &dataflow::ResultsCursor<'mir, 'tcx, MaybeInitializedLocals>,
+    borrowed: &dataflow::ResultsCursor<'mir, 'tcx, MaybeBorrowedLocals>,
+) {
+    // A local requires storage if it is initialized or borrowed. For now, a local
+    // becomes uninitialized if it is moved from, but is still considered "borrowed".
+    //
+    //     requires_storage := init | borrowed
+    //
+    // FIXME: This function is called in a loop, so it might be better to pass in a temporary
+    // bitset rather than cloning here.
+    let mut requires_storage = init.get().clone();
+    requires_storage.union(borrowed.get());
```

Comment:
actually, I think you can do better than this by intersecting with `storage_live ⋃ always_live_locals`, that's what the old code did.

Reply:
That's true. We should be using the actual `liveness` here, but I was worried about ill-formed MIR. I can try that in another PR, but for now `storage_live | always_live` is a fine approximation.

Reply:
Actually, once again `borrowed | init` is strictly more precise. `init` marks locals as uninitialized when we see `StorageDead`.

Reply:
Ah, and (crucially) so does `borrowed`, so you're right.